specification/security/resource-manager/Microsoft.Security/preview/2023-07-01-preview/integrations.json

```diff
@@ -0,0 +1,650 @@
+					"type": "number",
+					"readOnly": true
+				}
+			}
+		},
+		"IntegrationMetadata": {
+			"type": "object",
+			"description": "The integration metadata",
+			"properties": {
+				"createdBy": {
+					"description": "Integration Created by object id (GUID)",
+					"type": "string",
+					"readOnly": true
+				},
+				"createdOn": {
+					"description": "Integration creation date",
+					"type": "string",
+					"format": "date-time",
+					"readOnly": true
+				},
+				"updatedBy": {
+					"description": "Integration last updated by object id (GUID)",
+					"type": "string",
+					"readOnly": true
+				},
+				"updatedOn": {
+					"description": "Integration last update date",
+					"type": "string",
+					"format": "date-time",
+					"readOnly": true
+				}
+			}
+		},
+		"IntegrationAdditionalData": {
+			"type": "object",
+			"description": "The integration additional data (abstract)",
+			"discriminator": "integrationPlatform",
+			"properties": {
+				"integrationPlatform": {
+					"type": "string",
+					"description": "Describes the integration platform"
+				}
+			},
+			"required": [
+				"integrationPlatform"
+			]
+		},
+		"ServiceNowAdditionalData": {
+			"description": "Service Now specific additional data",
+			"allOf": [
+				{
+					"$ref": "#/definitions/IntegrationAdditionalData"
+				},
+				{
+					"properties": {
+						"instanceUrl": {
+							"description": "Integration creation date",
+							"type": "string"
+						},
+						"clientId": {
+							"description": "Integration creation date",
+							"type": "string"
+						},
+						"clientSecret": {
+							"description": "Integration creation date",
+							"type": "string"
+						},
+						"tableMetadata": {
+							"type": "array",
+							"items": {
+								"$ref": "#/definitions/ServiceNowTableMetadata"
+							}
+						}
+					}
+				}
+			],
+			"x-ms-discriminator-value": "ServiceNow"
+		},
+		"ServiceNowAuthenticationDetails": {
+			"description": "Service Now authentication details",
+			"type": "object",
+			"properties": {
+				"userName": {
+					"type": "string",
+					"description": "The integration user name"
+				},
+				"password": {
+					"type": "string",
+					"description": "The integration user password"
+				},
+				"clientId": {
+					"type": "string",
+					"description": "The integration client id"
+				},
+				"clientSecret": {
+					"type": "string",
+					"description": "The integration client secret"
+				},
+				"instanceUrl": {
+					"type": "string",
+					"description": "The integration instance url"
+				}
+			}
+		},
+		"ServiceNowMetadataResponse": {
+			"description": "Describes the Service Now authentication response and if successful, contains tables metadata response",
+			"type": "object",
+			"properties": {
+				"authenticationSucceeded": {
+					"type": "boolean",
+					"description": "Defines whether the Service Now authentication succeeded or not"
```

Comment:
For all new properties in this pr , consider updating the booleans to enums : 

[https://eng.ms/docs/products/arm/rp_onboarding/process/api_review_best_practices#common-issues-found-in-review](https://eng.ms/docs/products/arm/rp_onboarding/process/api_review_best_practices#common-issues-found-in-review)


Replace boolean/switch properties with better enum
A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].